contracts/root/staking/StakeManagerChildData.sol

```diff
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.19;
+
+/**
+ * @title StakeManagerChildData
+ * @notice Holds data to allow look-up between child chain manager contract address and child chain id.
+ * Note that this is contract is designed to be included in StakeManager. It is upgradeable.
+ */
+abstract contract StakeManagerChildData {
+    // Highest child chain id allocated thus far. Child chain id 0x00 is an invalid id.
+    uint256 internal counter;
+    // child chain id to child chain manager contract address.
+    mapping(uint256 => address) private managers;
+    // child chain manager contract address to child chain id.
+    mapping(address => uint256) private ids;
+
+    // Storage gap 
+    // solhint-disable-next-line var-name-mixedcase
+    uint256[1000] private __StorageGapStakeManagerChildData;
```

Comment:
Can we reduce this to something sensible like 50 and put it at the bottom, similar to OZ-style contracts?

Reply:
I would rename it to `__gap` as well to be consistent with the OZ implementations and to be compatible with libraries that handle upgrades

Reply:
The size doesn't matter too much: 10 should fingers crossed be enough. 50 is a lot. 1000 is more than anyone is likely to ever use. I will change it to 50.

I can move it to the bottom of the file. That will ensure it is after any other storage variables in the contract.

As for the name, if you create the storage layout file of the Solidity file (solc --storage-layout) you will find that having variables named __gap in each file can be quite confusing. I have found it to be better to have unique names in each file. In that way you can more quickly find the storage gap variable that has mistakenly not been modified when analysing an upgrade failure.